### PR TITLE
repo: add Travis CI, Codecov to flux-accounting

### PR DESCRIPTION
**Pull request by cmoussa1**
This PR adds Travis CI and Codecov to `flux-accounting`, as well as the badges indicating successful builds and the coverage percentage to the top of the **README**.

### CODE REVIEW

**Comment by cmoussa1** [comment]
Okay, now that I've added the **codecov.yml** file, I'll drop the [WIP] tag.

**Comment by cmoussa1** [comment]
Rebased to catch up after #42.

**Comment by dongahn** [comment]
@cmoussa1: this LGTM. @grondo?

File: .travis.yml
Comment (grondo) [inline .travis.yml:13]: The private codecov token for flux-accounting should not be checked in to your repo in the clear.

Instead set `CODECOV_TOKEN` as a "secret" environment variable in the Travis-CI settings page for flux-accounting:

https://travis-ci.com/github/flux-framework/flux-accounting/settings

Then that variable and your private codecov token will not be exposed to anyone except maintainers.

**Review by grondo — CHANGES_REQUESTED** [review]
Just onec comment about the private codecov token. This should be fixed before merging the PR.

File: .travis.yml
Comment (cmoussa1) [reply .travis.yml]: Thanks Mark. I believe I was able to set that environment variable correctly and pushed those changes up. Let me know if this is what you were thinking of. 

Looks like Codecov said that no coverage was uploaded, but looking at the [Error Reference](https://docs.codecov.io/docs/error-reference#section-missing-base-commit), I am wondering if it is because these changes haven't landed on the `master` branch yet?

Comment (grondo) [reply .travis.yml]: Yeah, since there is no coverage report for the main branch, codecov cannot report changes. That is expected.

**Review by grondo — APPROVED** [review]

File: .travis.yml
Comment (grondo) [inline .travis.yml:13]: You don't actually need to use `-t $CODECOV_TOKEN` here, just leave this off, the script will look for CODECOV_TOKEN.

Otherwise, LGTM!

Comment (cmoussa1) [reply .travis.yml]: Oh! My apologies. Just pushed a fix to remove that.

**Comment by grondo** [comment]
LGTM! Feel free to set MWP!

**Pull request merged**
